checker/response_body/response_body.go

```diff
@@ -33,8 +33,11 @@ func (c *ResponseBodyChecker) Check(t models.TestInterface, result *models.Resul
 			}
 			errs = append(errs, checkErrs...)
 		} else {
-			// compare bodies as simple strings
-			errs = append(errs, compare.Compare(expectedBody, result.ResponseBody, compare.CompareParams{})...)
+			// compare bodies as simple strings or check if regex matches
+			params := compare.CompareParams{
+				IsRegex: t.IsRegex(),
+			}
+			errs = append(errs, compare.Compare(expectedBody, result.ResponseBody, params)...)
```

Comment:
Hey, I'm not happy about the mess in Compare, so how about a small refactoring?

Let's split up the _deep compare_ and _plain string compare_ into separate functions. The one for strings comparing would accept the new IsRegexp parameter.

You can event put them into different files if you wish.

Reply:
Sure, I will do this